src/coordination/coordinator_instance.cpp

```diff
@@ -55,23 +56,41 @@ CoordinatorInstance::CoordinatorInstance()
                                            &CoordinatorInstance::MainFailCallback);
             });
 
-            std::ranges::for_each(repl_instances_, [this](auto &instance) {
-              instance.SetNewMainUUID(raft_state_.GetUUID());
-              instance.StartFrequentCheck();
-            });
+            std::ranges::for_each(repl_instances_, [](auto &instance) { instance.StartFrequentCheck(); });
+            is_shutting_down_ = false;
```

Comment:
What is this used for?

Reply:
When the coordinator needs to become a follower, we need to stop frequent checks as soon as possible. We can have the following issue. 
The coordinator is becoming a follower so we took lock to stop all the actions and stop threads. Other threads are starving and waiting to get the lock. This way we can check if we are actually in the shutting-down state so those threads can early and we can then shut down frequent checks completely.